framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1480,10 +1480,15 @@ public Void visitUnary(UnaryTree node, Void p) {
      * {@link #checkAssignability(AnnotatedTypeMirror, Tree)}.
      */
     @Override
-    public Void visitCompoundAssignment(CompoundAssignmentTree node, Void p) {
-        commonAssignmentCheck(node.getVariable(), node.getExpression(),
+    public Void visitCompoundAssignment(CompoundAssignmentTree compoundAssignmentTree, Void p) {
+        // Given a compound assignment:
+        // s += expr;
+        // Should be whether s + expr can be assigned to s,
```

Comment:
I had to read the `should be whether` part multiple times. Maybe the sentence can be clarified.